var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -30,11 +31,48 @@ class Papi(Package):
     version('5.4.1', '9134a99219c79767a11463a76b0b01a2')
     version('5.3.0', '367961dd0ab426e5ae367c2713924ffb')
 
+    variant('example', default=True, description='Install the example files')
+    variant('cuda', default=False, description='Enable CUDA support')
+    variant('nvml', default=False, description='Enable NVML support')
+    variant('infiniband', default=False, description='Enable Infiniband support')
+    variant('powercap', default=False, description='Enable powercap interface support')
+    variant('rapl', default=False, description='Enable RAPL support')
+    variant('lmsensors', default=False, description='Enable lm_sensors support')
+
+    conflicts('+cuda', when='@:5.6.0')
+    conflicts('+nvml', when='@:5.6.0')
+
+    depends_on('cuda', when='+cuda')
+    depends_on('cuda', when='+nvml')
+    depends_on('lm-sensors', when='+lmsensors')
+
     # Does not build with newer versions of gcc, see
     # https://bitbucket.org/icl/papi/issues/46/cannot-compile-on-arch-linux
     patch('https://bitbucket.org/icl/papi/commits/53de184a162b8a7edff48fed01a15980664e15b1/raw', sha256='64c57b3ad4026255238cc495df6abfacc41de391a0af497c27d0ac819444a1f8', when='@5.4.0:5.6.99%gcc@8:')
 
+    def setup_environment(self, spack_env, run_env):
+        if '+cuda' in self.spec or '+nvml' in self.spec:
+            spack_env.set('CUDA_DIR', self.spec['cuda'].prefix)
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.spec['cuda'].prefix,
+                                           "extras/CUPTI/lib64"))
+
     def install(self, spec, prefix):
+        if '+nvml' in spec:
+            with working_dir("src/components/nvml"):
+                configure_args = [
+                    "--with-nvml-incdir=%s/include" % spec['cuda'].prefix,
+                    "--with-nvml-libdir=%s/lib64/stubs" % spec['cuda'].prefix,
+                    "--with-cuda-dir=%s" % spec['cuda'].prefix]
+                configure(*configure_args)
+        if '+lmsensors' in spec:
+            with working_dir("src/components/lmsensors"):
+                configure_args = [
+                    "--with-sensors_incdir=%s/include/sensors" %
+                    spec['lm-sensors'].prefix,
+                    "--with-sensors_libdir=%s/lib64" %
```

Comment:
Will the libraries always be in `lib64`? We've had a lot of problems with this in the past across different systems. I would use `spec['lm-sensors'].libs.directories[0]` instead.

Reply:
OK, this required a modification to the lm-sensors package (included in the next set of changes).